python/common/python/math.java

```diff
@@ -0,0 +1,52 @@
+package python;
+
+@org.python.Module(
+        __doc__ = "A module that supplies some mathematical functions.\n"+
+                    "Current functions:\n"+
+                    "pow(x, y) -> Return x raised to the power y as type float."
+)
+
+public class math extends org.python.types.Module {
+    public math() {
+        super();
+    }
+
+    @org.python.Method(
+        __doc__ = "Returns x raised to the power y.\n",
+        args = {"x", "y"}
+    )
+
+    public static org.python.Object pow(org.python.Object x, org.python.Object y) {
+        double a;
+        double b;
+
+        if(x instanceof org.python.types.Int) {
+            a = ((org.python.types.Int) x).value;
+        } else if(x instanceof org.python.types.Float) {
+            a = ((org.python.types.Float) x).value;
+        } else {
+            throw new org.python.exceptions.TypeError("x needs to be of type int or float");
+        }
+
+        if(y instanceof org.python.types.Int) {
+            b = ((org.python.types.Int) y).value;
+        } else if(y instanceof org.python.types.Float) {
+            b = ((org.python.types.Float) y).value;
+        } else {
+            throw new org.python.exceptions.TypeError("y needs to be of type int or float");
+        }
+
+        if(a < 0 || b < 0) {
+            throw new org.python.exceptions.TypeError("Only values of x >= 0 and y >= 0 are accepted");
+        }
```

Comment:
To our understanding the python math pow should be able to take negative values for both the base and the exponent. Maybe look into this further.

Below is a part test of pow in a python3 shell
`
> import math
> pow(2, -3)
0.125
> pow(-2, 2)
4
> pow(-2, 2.5)
(1.7319121124709869e-15+5.656854249492381j)
>
 `